libnmstate/nm/device.py

```diff
@@ -49,6 +49,93 @@ def delete(dev):
         con_profile.delete()
 
 
+def reapply(dev, connection_profile):
+    """Reapply the given device or remote connection profile."""
+    mainloop = nmclient.mainloop()
+    mainloop.push_action(_safe_reapply_async, dev, connection_profile)
+
+
+def _safe_reapply_async(dev, connection_profile):
+    # Special cases exist (bugs) where reapply is not supported and
```

Comment:
AFAIU not all special cases are bugs unless you see it as a bug in the Linux kernel. Since reapply will fail in the other cases, the comment is a little bit misleading since we also fall back to apply when reapply fails. This also leads to the question why this special handling is needed at all. Why can this not be handled in the callback?